File: android/cpp/NativeEventDispatcher.cpp
#include "NativeEventDispatcher.h"

namespace RNSkiaVideo {
using namespace facebook;
using namespace jni;

jni::local_ref<NativeEventDispatcher>
NativeEventDispatcher::create(const RNSkiaVideo::JEventReceiver* receiver) {
  return newInstance((jlong)receiver);
}

void NativeEventDispatcher::registerNatives() {
  javaClassStatic()->registerNatives({makeNativeMethod(
      "nativeDispatchEvent", NativeEventDispatcher::dispatchEvent)});
}

void NativeEventDispatcher::dispatchEvent(alias_ref<JClass>, jlong receiverPtr,
                                          std::string eventName,
                                          alias_ref<jobject> data) {
  auto receiver = reinterpret_cast<JEventReceiver*>(receiverPtr);
  receiver->handleEvent(eventName, data);
}

} // namespace RNSkiaVideo

File: android/src/main/java/com/azzapp/rnskv/NativeEventDispatcher.java
package com.sheunglaili.rnskv;

public class NativeEventDispatcher {

  private long eventReceiver;


  public NativeEventDispatcher(long eventReceiver) {
    this.eventReceiver = eventReceiver;
  }

  public void dispatchEvent(String eventName, Object data) {
    nativeDispatchEvent(eventReceiver, eventName, data);
  }

  private static native void nativeDispatchEvent(long eventReceiver, String eventName, Object data);

}
